src/server/system_services/bucket_server.js

```diff
@@ -575,21 +582,24 @@ function get_bucket_changes_namespace(req, bucket, update_request, single_bucket
     if (!update_request.namespace.read_resources.length) throw new RpcError('INVALID_READ_RESOURCES');
 
     const read_resources = _.compact(update_request.namespace.read_resources
-        .map(ns_name => req.system.namespace_resources_by_name[ns_name] && req.system.namespace_resources_by_name[ns_name]._id));
+        .map(nsr => {
+                const res = req.system.namespace_resources_by_name[nsr.resource];
+                return res && { resource: res._id, path: nsr.path };
+        }));
     if (!read_resources.length || (read_resources.length !== update_request.namespace.read_resources.length)) {
         throw new RpcError('INVALID_READ_RESOURCES');
     }
     _.set(single_bucket_update, 'namespace.read_resources', read_resources);
-    const wr_obj = req.system.namespace_resources_by_name[update_request.namespace.write_resource];
-    const write_resource = wr_obj && wr_obj._id;
+    const wr_obj = req.system.namespace_resources_by_name[update_request.namespace.write_resource.resource];
+    const write_resource = wr_obj && { resource: wr_obj._id, path: update_request.namespace.write_resource.path };
     if (!write_resource) throw new RpcError('INVALID_WRITE_RESOURCES');
     _.set(single_bucket_update, 'namespace.write_resource', write_resource);
-    if (!_.includes(update_request.namespace.read_resources, update_request.namespace.write_resource)) {
+    if (!_.find(update_request.namespace.read_resources, update_request.namespace.write_resource)) {
```

Comment:
it's hard to notice that this works ok because _.find really knows how to match objects correctly. Maybe add a comment to explain?